src/home/saga.ts

```diff
@@ -1,51 +1,54 @@
 import { CleverTapInboxMessage, cleverTapInboxMessagesChannel } from 'src/home/cleverTapInbox'
 import { IdToNotification } from 'src/home/reducers'
-import { fetchCurrentRate } from 'src/localCurrency/actions'
+import { depositTransactionSucceeded } from 'src/jumpstart/slice'
+import { fetchLocalCurrencyRateSaga } from 'src/localCurrency/saga'
+import { fetchPositionsSaga, fetchShortcutsSaga } from 'src/positions/saga'
 import { executeShortcutSuccess } from 'src/positions/slice'
 import { withTimeout } from 'src/redux/sagas-helpers'
-import { fetchTokenBalances } from 'src/tokens/slice'
-import { updateTransactions } from 'src/transactions/slice'
+import { Actions as SendActions } from 'src/send/actions'
+import { swapSuccess } from 'src/swap/slice'
+import { fetchTokenBalancesSaga } from 'src/tokens/saga'
+import { updateFeedFirstPage, updateTransactions } from 'src/transactions/slice'
 import Logger from 'src/utils/Logger'
 import { safely } from 'src/utils/safely'
-import { getConnectedAccount } from 'src/web3/saga'
-import { call, cancelled, put, spawn, take, takeLeading } from 'typed-redux-saga'
+import { all, call, cancelled, put, spawn, take, takeLeading } from 'typed-redux-saga'
 
 const REFRESH_TIMEOUT = 15000
 const TAG = 'home/saga'
 
-export function withLoading<Fn extends (...args: any[]) => any>(fn: Fn, ...args: Parameters<Fn>) {
-  return function* withLoadingGen() {
-    yield* put(setLoading(true))
-    try {
-      const res = yield* call(fn, ...args)
-      return res
-    } finally {
-      yield* put(setLoading(false))
-    }
-  }
-}
-
-export function* refreshBalances() {
-  Logger.debug(TAG, 'Fetching all balances')
-  yield* call(getConnectedAccount)
-  yield* put(fetchTokenBalances({ showLoading: false }))
-  yield* put(fetchCurrentRate())
+export function* refreshAllBalances() {
+  yield* all([
+    call(fetchTokenBalancesSaga),
+    call(fetchLocalCurrencyRateSaga),
+    call(fetchPositionsSaga),
+    call(fetchShortcutsSaga),
+  ])
 }
 
 export function* watchRefreshBalances() {
-  yield* call(refreshBalances)
   yield* takeLeading(
-    [Actions.REFRESH_BALANCES, executeShortcutSuccess.type],
-    safely(withLoading(withTimeout(REFRESH_TIMEOUT, refreshBalances)))
+    [
+      updateTransactions.type, // emitted on recieve new transactions from graphql endpoint
+      updateFeedFirstPage.type, // emitted on recieve new transactions from zerion endpoint
+      Actions.REFRESH_BALANCES, // emitted on manual pull to refresh and app foregrounded
+      AccountActions.INITIALIZE_ACCOUNT_SUCCESS, // emitted after onboarding
+      // the below actions are emitted after a successful transaction that is
+      // initiated from the wallet.
+      executeShortcutSuccess.type,
+      depositSuccess.type,
+      withdrawSuccess.type,
+      SendActions.SEND_PAYMENT_SUCCESS,
+      createFiatConnectTransferCompleted.type,
+      depositTransactionSucceeded.type,
+      swapSuccess.type,
+    ],
```

Comment:
this is the main change

Reply:
does it belong to home? what do you think about moving it to tokens and spawning it in the root saga instead?

Reply:
yeah i had the same thought but i am not sure where is best for this. i also don't think tokens is the best place, because the balance refresh also involves refreshing local currency + positions. `home` kind of makes sense if we think of it being a middle ground between everything?

Reply:
yes, tokens is not the best place, but home doesn't feel right either for the same reason, because this saga updates the other screens too. i feel like it should be a separate one spawned from the root because of its "general" nature. don't have any good naming idea though 😅. so i agree that `home` is a reasonable place for now.